chef/automate
# User Detail Page Updates

### :nut_and_bolt: Description: What code changed, and why?
Updating the User Detail pages to more match the other pages. 
This is changing the below two pages:
* Profile page 
![image](https://user-images.githubusercontent.com/1679247/72654799-f9852280-3945-11ea-9917-1689ec0fddf5.png) 
* Admin users page (Settings -> Users). 

### :chains: Related Resources
* https://github.com/chef/automate/issues/2626
* https://github.com/chef/automate/issues/2625
* https://github.com/chef/automate/issues/2493

### :+1: Definition of Done
* Remove delete functionality from the detail page
* Match styles to the designs
* Add 2 tabs: Details, and Reset Password
* Change user notification text on reset password to: _Password reset for user: <user's username>.
    Fix the bug that does not update the user's display name in the user name after an edit has been made (requires a browser refresh, but shouldn't)
* Update left-nav to match designs
    * Heading: _USER MENU_
    * Item: _Profile_
* Update profile page to match designs
    * Heading: _Profile: <user's display name>_
    * Content: label of _Username_ and display the user's username
    * Two tabs: _Details_ and _Reset Password_ that function just like the user detail page

### :athletic_shoe: How to Build and Test the Change
1. `build components/automate-ui && start_all_services`
1. Log in as admin https://a2-dev.test
1. Go to the Admin users page https://a2-dev.test/settings/users
1. Create a new user with ID "bob"
1. Go to bob's user detail page https://a2-dev.test/settings/users/bob
1. Change the Display Name and ensure the "Save" button is only enable when there are changes in the text box. 
1. Save a change to the display name and ensure that a notification appears next to the "Save" button and not in the banner. 
1. Ensure the change was saved by refreshing the page. 
1. Switch to the "Reset Password" tab.
1. Change the password and ensure the "Reset Password" button is only enabled when the password is valid. 
1. Press the "Reset Password" and ensure a notification is displayed in the banner at the top. 
1. Log out and log in as bob. 
1. Go to the Profile page from the user menu at the top right. ![image](https://user-images.githubusercontent.com/1679247/72655289-b11b3400-3948-11ea-97b0-eeb7667a08fd.png)
![image](https://user-images.githubusercontent.com/1679247/72654799-f9852280-3945-11ea-9917-1689ec0fddf5.png) 
1. Do the same tests as the admin users page. Test updating the display name and resetting the password. The notification should work the same. 

### :white_check_mark: Checklist


### :camera: Screenshots, if applicable
User's Profile Page Reset Password
![image](https://user-images.githubusercontent.com/1679247/72655365-1838e880-3949-11ea-8019-496b0b1cf94a.png)
User's Profile Page Details
![image](https://user-images.githubusercontent.com/1679247/72655381-30106c80-3949-11ea-9762-a4a34c8c4ed0.png)
Admin Users Detail Page 
![image](https://user-images.githubusercontent.com/1679247/72655399-49b1b400-3949-11ea-9343-6ae25c689f77.png)
Admin User Detail Page Reset Password
![image](https://user-images.githubusercontent.com/1679247/72655406-546c4900-3949-11ea-99dc-ba4d4657b1c9.png)
